File: src/assignment3/src/controller_node.cpp
#include "ros/ros.h"
#include "assignment3/ExecutePlan.h"

bool execute_plan(assignment3::ExecutePlan::Request &req, assignment3::ExecutePlan::Response &res) {
    // Execute the plan by sending commands to the environment
    // Simulate success for now
    res.success = true;
    return true;
}

int main(int argc, char **argv) {
    ros::init(argc, argv, "controller_node");
    ros::NodeHandle n;

    ros::ServiceServer service = n.advertiseService("execute_plan", execute_plan);
    ros::spin();

    return 0;
}

File: src/assignment3/scripts/controller_node.py
import rospy
from assignment3.srv import ExecutePlan, ExecutePlanResponse

def execute_plan(req):
    # Execute the plan by sending commands to the environment
    # Simulate success for now
    success = True
    return ExecutePlanResponse(success)

def controller_server():
    rospy.init_node('controller_node')
    service = rospy.Service('execute_plan', ExecutePlan, execute_plan)
    rospy.spin()

if __name__ == "__main__":
    controller_server()
